cache.go

```diff
@@ -17,27 +18,65 @@ const (
 	cacheMountUnset   = ""
 
 	BazelDefaultSocketID = "bazel-default" // Default ID for bazel socket
+
 )
 
+// getSccacheSource returns a Source for downloading and verifying sccache using SourceHTTP
+func getSccacheSource(p *ocispecs.Platform) Source {
+	// sccache version and download configuration
+	const (
+		sccacheVersion     = "v0.10.0"
+		sccacheDownloadURL = "https://github.com/mozilla/sccache/releases/download"
+	)
+
+	// Determine architecture string and checksum based on platform
+	var arch, checksum string
+	switch {
+	case p.Architecture == "amd64":
+		arch = "x86_64-unknown-linux-musl"
+		checksum = "1fbb35e135660d04a2d5e42b59c7874d39b3deb17de56330b25b713ec59f849b"
+	case p.Architecture == "arm64":
+		arch = "aarch64-unknown-linux-musl"
+		checksum = "d6a1ce4acd02b937cd61bc675a8be029a60f7bc167594c33d75732bbc0a07400"
+	case p.Architecture == "arm" && p.Variant == "v7":
+		arch = "armv7-unknown-linux-musleabi"
+		checksum = "ab7af4e5c78aa71f38145e7bed41dd944d99ce5a00339424d927f8bbd8b61b78"
+	default:
+		// Fallback to linux x64 for unsupported platforms
+		arch = "x86_64-unknown-linux-musl"
+		checksum = "1fbb35e135660d04a2d5e42b59c7874d39b3deb17de56330b25b713ec59f849b"
+	}
+
+	// Build download URL
+	url := fmt.Sprintf("%s/%s/sccache-%s-%s.tar.gz", sccacheDownloadURL, sccacheVersion, sccacheVersion, arch)
+
+	return Source{
+		HTTP: &SourceHTTP{
+			URL:    url,
+			Digest: digest.Digest("sha256:" + checksum),
+		},
+	}
+}
+
 // CacheConfig configures a cache to use for a build.
-//
-// Other, cache types may be added in the future, such as:
-// - rust compiler cache
-// ...
 type CacheConfig struct {
 	// Dir specifies a generic cache directory configuration.
 	Dir *CacheDir `json:"dir,omitempty" yaml:"dir,omitempty" jsonschema:"oneof_required=dir"`
 	// GoBuild specifies a cache for Go's incremental build artifacts.
 	// This should speed up repeated builds of Go projects.
 	GoBuild *GoBuildCache `json:"gobuild,omitempty" yaml:"gobuild,omitempty" jsonschema:"oneof_required=gobuild"`
+	// CargoBuild specifies a cache for Rust/Cargo build artifacts.
+	// This uses sccache to cache Rust compilation artifacts.
+	CargoBuild *CargoSCCache `json:"cargosccache,omitempty" yaml:"cargosccache,omitempty" jsonschema:"oneof_required=cargosccache"`
```

Comment:
nit on the naming here:

1. `CargoBuild` is inconsistent with the rest.
2. The type could probably just be called `SCCache`
3. The actual field names for go and json/yaml maybe should be rustsccache since they aren't related to cargo at all.